mesonbuild/linkers.py

```diff
@@ -801,18 +811,21 @@ class MSVCDynamicLinker(VisualStudioLikeLinkerMixin, DynamicLinker):
 
     """Microsoft's Link.exe."""
 
-    def __init__(self, for_machine: mesonlib.MachineChoice,
-                 *, version: str = 'unknown version'):
-        super().__init__(['link.exe'], for_machine, 'link', version=version)
+    def __init__(self, for_machine: mesonlib.MachineChoice, *,
+                 exelist: typing.List[str] = ['link.exe'], prefix: str = '',
```

Comment:
__[Flake8]__ [B006] Do not use mutable data structures for argument defaults.  They are created during function definition time. All calls to the function reuse this one instance of that data structure, persisting changes between them. ([link](https://sider.review/gh/repos/19784232/pulls/5856#issue-4487207))

<sub>posted by Sider</sub>